### PR TITLE
IDEX-2560: fixed wrong deletion index of files with similar path

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java [hunk 1]
```diff
@@ -188,9 +191,9 @@ public void onRenamed() {
                         StringMap<EditorPartPresenter> editors = editorAgent.getOpenedEditors();
                         Array<EditorPartPresenter> openedEditors = editors.getValues();
 
-                        for (EditorPartPresenter editor : openedEditors.asIterable()) {
-                            partStackPresenter.removePart(editor);
-                        }
+//                        for (EditorPartPresenter editor : openedEditors.asIterable()) {
+//                            partStackPresenter.removePart(editor);
+//                        }
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java]: ?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java [hunk 2]
```diff
@@ -270,6 +273,7 @@ private void checkOpenedFiles(ItemReference itemBeforeRenaming, String newName)
                 protected void onSuccess(Array<ItemReference> result) {
                     StringMap<ItemReference> children = Collections.createStringMap();
                     for (ItemReference itemReference : result.asIterable()) {
+                        Log.info(getClass(), "server get index " + itemReference.getPath());
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java]: Do we need that?

File: platform-api/che-core-api-vfs/src/main/java/org/eclipse/che/api/vfs/server/search/LuceneSearcher.java [hunk 3]
```diff
@@ -218,13 +218,55 @@ protected void addFile(VirtualFile virtualFile) throws ServerException {
     }
 
     @Override
-    public final void delete(String path) throws ServerException {
-        doDelete(new Term("path", path));
+    public final void delete(String path, boolean isFile) throws ServerException {
+        LOG.info("===============================================delete worked" + path);
+        String[] result = search((new QueryExpression()).setPath("/"));
+        LOG.info("Before *************************************************************");
```
Comment (skabashnyuk) [inline platform-api/che-core-api-vfs/src/main/java/org/eclipse/che/api/vfs/server/search/LuceneSearcher.java]: ?

File: platform-api/che-core-api-vfs/src/main/java/org/eclipse/che/api/vfs/server/search/LuceneSearcher.java [hunk 4]
```diff
@@ -218,13 +218,55 @@ protected void addFile(VirtualFile virtualFile) throws ServerException {
     }
 
     @Override
-    public final void delete(String path) throws ServerException {
-        doDelete(new Term("path", path));
+    public final void delete(String path, boolean isFile) throws ServerException {
+        LOG.info("===============================================delete worked" + path);
```
Comment (skabashnyuk) [inline platform-api/che-core-api-vfs/src/main/java/org/eclipse/che/api/vfs/server/search/LuceneSearcher.java]: ?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java (hunk 1 above)
Comment (AndrienkoAleksandr) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RenameItemAction.java]: These lines was commented for testing updating reference inside files

**Pull request closed without merging**